src/main/java/com/banking/app/bankingApp/controller/TransactionController.java

```diff
@@ -1,5 +1,80 @@
 package com.banking.app.bankingApp.controller;
 
+import com.banking.app.bankingApp.request.transactions.CreateTransaction;
+import com.banking.app.bankingApp.request.transactions.UpdateTransaction;
+import com.banking.app.bankingApp.response.transactions.Transaction;
+import com.banking.app.bankingApp.service.transactions.TransactionManagementService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.NoResultException;
+import java.util.List;
+
+@RestController
 public class TransactionController {
+    private TransactionManagementService transactionManagementService;
+
+    public TransactionController() {
+        transactionManagementService = new TransactionManagementService();
+    }
+
+    @PostMapping(value = "/transactions")
+    public ResponseEntity<Transaction> addTransaction(@RequestBody CreateTransaction createTransaction) {
+        try {
+            Transaction transaction = transactionManagementService.addTransaction(createTransaction);
+            return ResponseEntity.status(HttpStatus.OK).body(transaction);
+        } catch (IllegalStateException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        } catch (RuntimeException e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GetMapping(value = "/transactions/{id}")
+    public ResponseEntity<Transaction> getTransactionsByTransactionId(@PathVariable("id") String id) {
+        try {
+            Transaction transaction = transactionManagementService.getTransactionById(id);
+            return ResponseEntity.status(HttpStatus.OK).body(transaction);
+        } catch (NoResultException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        } catch (RuntimeException e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GetMapping(value = "/transactions/{id}")
```

Comment:
You have 2 methods with same route /transactions/{id} and http method. How will controller know where to go?